### PR TITLE
feat: More checks for return types from GraphQL operations

### PR DESCRIPTION
**Pull request by jmartisk**
a
b

### CODE REVIEW

**Comment by angelozerr** [comment]
@jmartisk many thanks for your contribution!

Could youp please add a screenshot in your PR which shows in action your contribution. Thanks!

**Comment by jmartisk** [comment]
@angelozerr done, added two screenshots in the description

File: src/main/java/com/redhat/devtools/intellij/lsp4mp4ij/psi/internal/graphql/MicroProfileGraphQLConstants.java [hunk 1]
```diff
@@ -23,10 +23,12 @@ private MicroProfileGraphQLConstants() {
 
     public static final String QUERY_ANNOTATION = "org.eclipse.microprofile.graphql.Query";
     public static final String MUTATION_ANNOTATION = "org.eclipse.microprofile.graphql.Mutation";
-    public static final String SUBSCRIPTION_ANNOTATION = "org.eclipse.microprofile.graphql.Subscription";
```
Comment (fbricon) [inline src/main/java/com/redhat/devtools/intellij/lsp4mp4ij/psi/internal/graphql/MicroProfileGraphQLConstants.java:26]: keep it and rename MICROPROFILE_SUBSCRIPTION_ANNOTATION

File: src/main/java/com/redhat/devtools/intellij/lsp4mp4ij/psi/internal/graphql/MicroProfileGraphQLConstants.java [hunk 2]
```diff
@@ -23,10 +23,12 @@ private MicroProfileGraphQLConstants() {
...
+    public static final String SUBSCRIPTION_ANNOTATION = "io.smallrye.graphql.api.Subscription";
```
Comment (fbricon) [inline src/main/java/com/redhat/devtools/intellij/lsp4mp4ij/psi/internal/graphql/MicroProfileGraphQLConstants.java:26]: SMALLRYE_SUBSCRIPTION_ANNOTATION

**Review by fbricon — CHANGES_REQUESTED** [review]
We should keep supporting the existing `org.eclipse.microprofile.graphql.Subscription`

File: src/main/java/com/redhat/devtools/intellij/lsp4mp4ij/psi/internal/graphql/MicroProfileGraphQLConstants.java (hunk 1 above)
Comment (jmartisk) [reply src/main/java/com/redhat/devtools/intellij/lsp4mp4ij/psi/internal/graphql/MicroProfileGraphQLConstants.java:26]: Actually there is no `org.eclipse.microprofile.graphql.Subscription` class (see https://github.com/eclipse/microprofile-graphql/tree/main/server/api/src/main/java/org/eclipse/microprofile/graphql) - that was an error, thus changing the class name isn't really a breaking change.

Comment (angelozerr) [reply src/main/java/com/redhat/devtools/intellij/lsp4mp4ij/psi/internal/graphql/MicroProfileGraphQLConstants.java:26]: @datho7561 do you remember why we use this class `org.eclipse.microprofile.graphql.Subscription`  in JDT?

Comment (jmartisk) [reply src/main/java/com/redhat/devtools/intellij/lsp4mp4ij/psi/internal/graphql/MicroProfileGraphQLConstants.java:26]: I think it was my mistake, adding it via https://github.com/redhat-developer/intellij-quarkus/commit/d7a9ec706bfd90409f1b8be4df477543bf1186cd#diff-0800a063c095ddbc177c8d186ae4dcbe5299b3b65d6507f0541bbe6be2aa3084R26

**Comment by jmartisk** [comment]
Hi @fbricon I noticed this has been waiting for a long time, could you give it another look, and let me know if anything else is needed before merging? Thanks

**Comment by fbricon** [comment]
Thanks @jmartisk, sorry for the delay in handling this one.

**Pull request merged**